src/datastore/parser.cpp

```diff
@@ -0,0 +1,166 @@
+        std::cout << "Is deleted?:  " << (*it)[15] << std::endl;
+        std::cout << "---Souvenir:  " << (*it)[16] << std::endl;
+        std::cout << "------Price:  " << (*it)[17] << std::endl;
+        std::cout << "Souvenir ID:  " << (*it)[18] << std::endl;
+        std::cout << "Is deleted?:  " << (*it)[19] << std::endl;
+        std::cout << "---Souvenir:  " << (*it)[20] << std::endl;
+        std::cout << "------Price:  " << (*it)[21] << std::endl;
+        std::cout << "Souvenir ID:  " << (*it)[22] << std::endl;
+        std::cout << "Is deleted?:  " << (*it)[23] << std::endl;
+        std::cout << "---Souvenir:  " << (*it)[24] << std::endl;
+        std::cout << "------Price:  " << (*it)[25] << std::endl;
+        std::cout << "Souvenir ID:  " << (*it)[26] << std::endl;
+        std::cout << "Is deleted?:  " << (*it)[27] << std::endl;
+        std::cout << "---Souvenir:  " << (*it)[28] << std::endl;
+        std::cout << "------Price:  " << (*it)[29] << std::endl;
+        std::cout << "Souvenir ID:  " << (*it)[30] << std::endl;
+        std::cout << "Is deleted?:  " << (*it)[31] << std::endl;
+        std::cout << "---Souvenir:  " << (*it)[32] << std::endl;
+        std::cout << "------Price:  " << (*it)[33] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+void distanceParseDebug(std::string path)
+{
+    std::vector<std::vector<std::string>> distanceData = loadDistanceData(path);
+    for (std::vector<std::vector<std::string>>::iterator it = distanceData.begin(); it != distanceData.end(); ++it)
+    {
+
+        std::cout << "Stadium ID:   " << (*it)[0] << std::endl;
+        std::cout << "Stadium Name: " << (*it)[1] << std::endl;
+        std::cout << "Stadium ID:   " << (*it)[2] << std::endl;
+        std::cout << "Stadium Name: " << (*it)[3] << std::endl;
+        std::cout << "Distance:     " << (*it)[4] << std::endl;
+        std::cout << std::endl;
+    }
+}
+
+std::vector<std::vector<std::string>> loadTeamData(std::string path)
+{
+    // Initializing - vector will hold all lines of data within the document
+    std::vector<std::vector<std::string>> allRows;
+    std::ifstream infile(path, std::ios::in);
+    int lineCount = 0;
+
+    // Checks if file is open if now throw error
+    if (infile.is_open())
+    {
+        std::string line;
+        while (infile.good())
+        {
+            std::getline(infile, line);
+            if (!infile.eof())
+            {
+                if (line[0] == '#' || line[0] == ' ')
+                {
+                    continue; // ignore comment lines and spaces
+                }
+
+                ++lineCount;
+
+                std::vector<std::string> columns;
+
+                // used for seperating words
+                std::stringstream row(line);
+                std::string word;
+                int wordCount(0);
+
+                while (!row.eof())
+                {
+                    if (wordCount < 5 || wordCount > 6)
```

Comment:
Why are we hard-coding a file parser for `.csv` files?

Reply:
because there are quotes at a specific point

Reply:
It's not like we don't know what each column should be, I just happen to tell it to extract the info differently once you reach the lines with quotes

Reply:
Read my review description. The main point is that there is going to be redundant code. And its easily fixed. I posted a sample long ago
```cpp
std::ifstream fin("somefile.csv"); //replace with std::stringstream since you're using that
//...
std::string data;
std::getline(fin, data, ',');

//If the first character is a quotation mark
if(data[0] == '"')
{
    std::string extra;

    //Input again and stop if it ends with a quotation mark
    do
    {
        std::getline(fin, extra, ',');
        data += extra;
    } while(extra.endsWith('"'));
}
```
Just because we know something is a certain way now, doesn't mean it will always stay the same in the future.